refinery/tool_manager/utils.py

```diff
@@ -108,6 +110,56 @@ def create_tool_definition(annotation_data):
 
 
 @transaction.atomic
+def create_tool(tool_launch_configuration, user_instance):
+    """
+   :param tool_launch_configuration: dict of data that represents a Tool
+   :param user_instance: User object that made the request to create said Tool
+   :returns: The created Tool object
+   """
+    # NOTE: that the usual exceptions for the get() aren't handled because
+    # we're in the scope of an atomic transaction
+    tool_definition = ToolDefinition.objects.get(
+        uuid=tool_launch_configuration["tool_definition_uuid"]
+    )
+    tool = ToolFactory(
+        name="{}-launch".format(tool_definition.name),
+        tool_definition=tool_definition,
+        file_relationships=tool_launch_configuration["file_relationships"]
+    )
+    try:
+        tool.parameters = tool_launch_configuration["parameters"]
+    except KeyError:
+        # parameters are not required for Tools to launch properly
+        pass
+
+    if tool.get_tool_type() == ToolDefinition.VISUALIZATION:
+        try:
+            tool.output_files = tool_launch_configuration["output_files"]
+        except KeyError:
+            # output_files aren't required for Vis Tools
+            pass
+
+        # Create a unique container name that adheres to docker's specs
+        tool.container_name = "{}-{}".format(
+            tool.name.replace(" ", ""),
+            tool.uuid
+        )
+
+    if tool.get_tool_type() == ToolDefinition.WORKFLOW:
+        try:
+            tool.output_files = tool_launch_configuration["output_files"]
+        except KeyError:
```

Comment:
Yes in tests, Since RuntimeErros can be raised in many places during tool creation, its helpful to have a message associated so that I can assert that the proper exception is being hit.